plugins/jobtype/src/azkaban/jobtype/HadoopSparkJob.java

```diff
@@ -530,6 +521,46 @@ private static void handleStandardArgument(Props jobProps,
   }
 
   /**
+   * This method is used to get spark home from plugin's jobtype config.
+   * If spark.{sparkVersion}.home is set in commonprivate.properties/private.properties, then that will be returned.
+   * If spark.{sparkVersion}.home is not set and spark.home.dir is set then it will retrieve Spark directory inside
+   * spark.home.dir, matching spark.home.prefix + sparkVersion pattern. Regex pattern can be passed as properties for
+   * version formatting.
+   * @param sparkVersion
+   * @return
+   */
+  private String getSparkHome(String sparkVersion) {
+    String sparkHome = getSysProps().get("spark." + sparkVersion + ".home");
+    if (sparkHome == null) {
+      info("Couldn't find spark." + sparkVersion + ".home property.");
+      String sparkDir = getSysProps().get("spark.home.dir");
```

Comment:
Should probably rename "spark.home.dir" to spark.base.dir" to be more accurate about the purpose of these configurations.